components/event-publisher-proxy/pkg/handler/handler.go

```diff
@@ -69,13 +87,68 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 		h.writeResponse(writer, http.StatusMethodNotAllowed, nil)
 		return
 	}
+	// Limit server from reading a huge payload
+	request.Body = http.MaxBytesReader(writer, request.Body, h.Options.MaxRequestSize)
+	uri := request.RequestURI
+
+	// Process /publish endpoint
+	// Gets a CE and sends it to BEB
+	if isARequestWithCE(uri) {
+		h.publishCloudEvents(writer, request)
+		return
+	}
 
-	// validate request URI
-	if request.RequestURI != "/publish" {
-		h.writeResponse(writer, http.StatusNotFound, nil)
+	// Process /:application/v1/events
+	// Publishes a legacy event as CE v1.0 to BEB
+	if isARequestWithLegacyEvent(uri) {
+		h.publishLegacyEventsAsCE(writer, request)
 		return
 	}
 
+	h.writeResponse(writer, http.StatusNotFound, nil)
+	return
+}
+
+func isARequestWithCE(uri string) bool {
+	if uri == publishEndpoint {
+		return true
+	}
+	return false
+}
+func isARequestWithLegacyEvent(uri string) bool {
+	// Assuming the path should be of the form /:application/v1/events
+	uriPathSegments := make([]string, 0)
+
+	for _, segment := range strings.Split(uri, "/") {
+		if strings.Trim(segment, " ") != "" {
```

Comment:
I recommend using `TrimSpace` to trim all whitespaces (e.g. `\t`...).

Reply:
Changed